src/FSM.c

```diff
@@ -244,12 +261,149 @@ extern void FSM(void *dummy){
 
 		//TMP command
 		else if(strcmp(commandString, "TMP") == 0){
-
 			// Get motor temperature
 			// Send out temperature
 
 		}
 
+		//TMx Command
+		else if(strncmp(commandString, "TM", 2) == 0 && strlen(commandString) == TM_COMMAND_LENGTH){
+			uint16_t temperature;
+			ADC_sensors_t ADC_sensor;
+			switch(commandString[TM_MOTOR_NUMBER_LOCATION]){
```

Comment:
This switch statement is also not necessary. It could be a "if('A') else" statement (not else if). E.g., you could convert the character '1' to an integer (see Poorna's code in Motor Controller 2.0 fsm.c) and add (Temp_ADC1 - 1) to the result to get the enum for Temp_ADC1. For '2', you convert to integer, add (Temp_ADC1 - 1) (same as for '1') and get the enum for Temp_ADC2. This will work for all the way up to '8', hence you don't need a switch statement, just an if...else. Let me know if you have any questions here.